### PR TITLE
feat(landing): Update the landing page to use maplibre

### PR DESCRIPTION
**Pull request by Wentao-Kuang**

### CODE REVIEW

File: packages/landing/src/map.ts [hunk 1]
```diff
@@ -53,121 +47,105 @@ export class Basemaps {
...
 
     /** Import location and rendering information from the url */
     private updateFromUrl(): void {
         this.config = WindowUrl.fromUrl(window.location.search);
-        const tileMatrix = this.config.tileMatrix;
         const location = this.getLocationFromHash();
-
-        const loc = this.locationFromLonLat(location);
         const tileGrid = getTileGrid(this.config.tileMatrix.identifier);
+        const projection = Projection.get(this.config.tileMatrix.projection);
```
Comment (Wentao-Kuang) [inline packages/landing/src/map.ts:64]: After adding this line, I cannot build the index.js in yarn bundle

File: packages/landing/src/url.ts [hunk 2]
```diff
@@ -99,6 +102,10 @@ export const WindowUrl = {
             return `${baseTileUrl}/{z}/{x}/{y}.${WindowUrl.ImageFormat}${api}`;
         }
 
+        if (urlType === MapOptionType.VectorTile) {
```
Comment (blacha) [inline packages/landing/src/url.ts:105]: if this is `VectorTile` then we should update `Tile` to `RasterTile` 

I prefer Yoda style, `TileVector` `TileRaster`

File: packages/landing/src/tile.matrix.ts [hunk 3]
```diff
@@ -54,6 +55,27 @@ export class TileGrid {
         });
     }
 
+    getStyle(config: MapOptions): Style | string {
+        if (config.imageId === 'topographic') return WindowUrl.toTileUrl(config, MapOptionType.VectorTile);
+        return {
+            version: 8,
+            sources: {
+                'raster-tiles': {
```
Comment (blacha) [inline packages/landing/src/tile.matrix.ts:63]: would be easier if this was not `-` eg `raster` or `basemaps` then you wont have to quote it.

File: packages/landing/src/tile.matrix.ts [hunk 4]
```diff
@@ -75,3 +99,17 @@ export function getTileGrid(id: string): TileGrid {
     }
     return GoogleTileGrid;
 }
+
+export function getNZTMTile(tileMatrix: TileMatrixSet, location: MapLocation): Tile {
+    const projection = Projection.get(tileMatrix);
+    const nztmCoords = projection.fromWgs84([location.lon, location.lat]);
+    const nztmCenter = tileMatrix.sourceToPixels(nztmCoords[0], nztmCoords[1], location.zoom);
+    const nztmTile = { x: nztmCenter.x / 256, y: nztmCenter.y / 256, z: location.zoom };
```
Comment (blacha) [inline packages/landing/src/tile.matrix.ts:107]: `256` is a actually `tileMatrix.tileSize` (or something)

File: packages/landing/src/tile.matrix.ts [hunk 5]
```diff
@@ -75,3 +99,17 @@ export function getTileGrid(id: string): TileGrid {
     }
     return GoogleTileGrid;
 }
+
+export function getNZTMTile(tileMatrix: TileMatrixSet, location: MapLocation): Tile {
```
Comment (blacha) [inline packages/landing/src/tile.matrix.ts:103]: this is not specific to NZTM, this `mapboxToTileMatrixLocation`

File: packages/landing/src/attribution.ts [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+    filteredRecords: AttributionCollection[] = [];
+    attributionControl: mapboxgl.AttributionControl;
+
+    /**
+     * Initialize monitoring the OpenLayers map and set the source attributions when changed.
+     */
+    static init(map: mapboxgl.Map, config: MapOptions): void {
+        const attribution = new MapAttribution(map, config);
+
+        map.on('load', (): void => {
```
Comment (blacha) [inline packages/landing/src/attribution.ts:36]: could these just be  `map.on('load',  () => attribution.updateAttribution())`

File: packages/landing/src/map.ts [hunk 7]
```diff
@@ -53,115 +29,54 @@ export class Basemaps {
...
-
-    tileTrackTimer: unknown | null = null;
-    trackTileLoad = (evt: BaseEvent): boolean => {
-        if (!isTileLoadEvent(evt)) return true;
-        const metricName = 'tile:' + evt.tile.getTileCoord().join('-');
-        if (evt.type === 'tileloadstart') {
-            this.tileTimer.set(metricName, Date.now());
-            return true;
-        }
+        if (tileGrid.tileMatrix.identifier !== GoogleTms.identifier) this.map.setMaxBounds([-180, -85.06, 180, 85.06]);
```
Comment (blacha) [inline packages/landing/src/map.ts:58]: can we pull these hard coded values from the GoogleTms.extent or something?

Comment (Wentao-Kuang) [reply packages/landing/src/map.ts:58]: Not sure how to make this work now. leave it for next time.

**Review by blacha — APPROVED** [review]

**Pull request merged**